File: .claude/hooks/auto-allow-exit-plan.mjs
#!/usr/bin/env node
// @hookstack permission-request-auto-allow-exit-plan
// Auto-autorise la sortie du mode plan (PermissionRequest)
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';

export function run(input) {
  const toolName = input.tool_name ?? input.tool ?? '';
  if (toolName !== 'exit_plan_mode') return null;
  return {
    hookSpecificOutput: {
      hookEventName: 'PermissionRequest',
      decision: { behavior: 'allow' },
    },
  };
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result) process.stdout.write(JSON.stringify(result));
}

File: .claude/hooks/auto-allow-exit-plan.py
#!/usr/bin/env python3
# @hookstack permission-request-auto-allow-exit-plan
"""Auto-allows exiting plan mode (PermissionRequest)."""
import json
import sys


def run(input_data=None):
    tool_name = (input_data or {}).get("tool_name") or (input_data or {}).get("tool") or ""
    if tool_name != "exit_plan_mode":
        return None
    return {
        "hookSpecificOutput": {
            "hookEventName": "PermissionRequest",
            "decision": {"behavior": "allow"},
        },
    }


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result:
        sys.stdout.write(json.dumps(result))
